.buildkite/scripts/resolve_es_treeish.sh

```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-set -euo pipefail
+set +u
```

Comment:
Do we need to cancel failing on errors entirely, which is what this does?

My understanding of the problem is that if `SNAPSHOT` is undefined we fail here: https://github.com/elastic/logstash-filter-elastic_integration/blob/main/.buildkite/scripts/resolve_es_treeish.sh#L10

So what if we just either:

1) change to `set -eo pipefail`, which shouldn't cause failures anymore if undefined variables are used?

2) right before https://github.com/elastic/logstash-filter-elastic_integration/blob/main/.buildkite/scripts/resolve_es_treeish.sh#L10 we `set +u`?

Reply:
Our expectation is `SNAPSHOT=true` is explicitly set if we need to test against snapshot, otherwise, we use release version. That is said, in order to test against release versions, we don't need to set `SNAPSHOT=false`.

Reply:
Ok, so in this case, we can simply skip failing hard if `SNAPSHOT` is undefined, so I think it's enough to `set +u` (or `set +o nounset`, if you prefer the verbose version [man page](https://www.gnu.org/software/bash/manual/bash.html#The-Set-Builtin)) just before the conditional. If we do that, `set -euo pipefail` can remain at the top of the file and benefit us from the usual error checks for any other potential problem.